libs/SalesforceSDKCore/SalesforceSDKCore/Classes/UserAccount/SFUserAccountManager.m

```diff
@@ -1417,7 +1493,11 @@ - (void)setCurrentUserInternal:(SFUserAccount*)user {
                 [self willChangeValueForKey:@"currentUser"];
                 _currentUser = user;
                 [self setCurrentUserIdentity:user.accountIdentity];
-                if (user.credentials.domain)
+                
+                BOOL isNativeLogin = self.nativeLoginEnabled && !self.shouldFallbackToWebAuthentication;
+                // Native Login uses a secondary Connected App tied to a specifc community url.  If the
```

Comment:
Back to my earlier comment:
- web login does not require the app to programmatically pass the client id (it's read from bootconfig)
- native login expects the app to programmatically pass the client id
I wonder for consistency if we should have the native login client id also be defined in bootconfig?

Reply:
Makes sense. We can always add a bootconfig entry + method that does not take the client id later, if we think it has value.